pkg/tsdb/cloudwatch/metric_find_query.go

```diff
@@ -699,11 +699,15 @@ func (e *cloudWatchExecutor) handleGetAllLogGroups(pluginCtx backend.PluginConte
 	var response *cloudwatchlogs.DescribeLogGroupsOutput
 	result := make([]suggestData, 0)
 	for {
-		response, err = logsClient.DescribeLogGroups(&cloudwatchlogs.DescribeLogGroupsInput{
-			LogGroupNamePrefix: aws.String(logGroupNamePrefix),
-			NextToken:          nextToken,
-			Limit:              aws.Int64(defaultLogGroupLimit),
-		})
+		input := &cloudwatchlogs.DescribeLogGroupsInput{
+			Limit:     aws.Int64(defaultLogGroupLimit),
+			NextToken: nextToken,
+		}
+		if len(logGroupNamePrefix) > 0 {
+			input.LogGroupNamePrefix = aws.String(logGroupNamePrefix)
+		}
```

Comment:
Would you explain this change here? Was there something not working as intended with the original code?

Reply:
Yes, using an empty string for the `logGroupNamePrefix` causes the following error:

```
unexpected error InvalidParameter: 1 validation error(s) found.
- minimum field size of 1, DescribeLogGroupsInput.LogGroupNamePrefix.
```

I'm assuming the AWS SDK does a check for `nil` in which case the `LogGroupNamePrefix` is not validated. There's probably a better way of doing this, but it works for now.

Reply:
OK! Sounds like a different issue from the main issue here? This can be reported and fixed separately if you want.